lightly_studio/src/lightly_studio/core/dataset.py

```diff
@@ -288,11 +305,68 @@ def add_samples_from_path(
             image_paths=image_paths,
         )
 
+        if tag_depth == 1 and created_sample_ids:
+            self._tag_samples_by_directory(
+                input_path=path,
+                sample_ids=created_sample_ids,
+            )
+
         if embed:
             _generate_embeddings(
                 session=self.session, dataset_id=self.dataset_id, sample_ids=created_sample_ids
             )
 
+    def _tag_samples_by_directory(
+        self,
+        input_path: PathLike,
+        sample_ids: list[UUID],
+    ) -> None:
+        """Tags samples based on their first-level subdirectory relative to input_path."""
+
+        print(f"Adding directory tags to {len(sample_ids)} new samples.")
+        parent_dir_to_sample_ids = defaultdict(list)
```

Comment:
Minor: Can you move the `parent_dir_to_sample_ids` var closer to where it is used? E.g. directly before the for-loop?